apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -320,9 +333,19 @@ test_resource(#{type := Type} = Params) ->
         {ok, #resource_type{}} ->
             ResId = maps:get(id, Params, resource_id()),
             try
-                _ = create_resource(maps:put(id, ResId, Params)),
-                true = is_source_alive(ResId),
-                ok
+                case create_resource(maps:put(id, ResId, Params), no_retry) of
```

Comment:
nit , 335 and 336 to ```try .... of ```

Reply:
Try of is not safe as there can be case_clause exception